### ISSUE TITLE
Add a typechecking only target to the Makefile

### ISSUE THREAD

**Issue by gallais** [agda/agda#3939]
Thanks to 08522e5e3aa24b3ff56a3dfd1dd3b55a80f9c759 the edit / compile cycle has gotten a lot of faster.

We could go even faster in the cases we are only interested in fixing all of
the type errors before even attempting to generate code.

Reading haskell/cabal#1176, it seems that using the option
`--ghc-options="-fno-code -fwrite-interface"` should do the trick.

* `no-code` skips the code generation step
* `write-interface` still generates the `.hi` files so that we have some form of caching for this process

And it does work with a mere `cabal install`. However if I try to add a target
to `Makefile` mimicking `quicker-install-bin` it fails: `cabal` tries to build
`Setup.hs` (I guess?) but the `no-code` option means nothing is generated
which leads to the next step failing.

We could maybe find a way to generate that first file and then call `cabal`
using `no-code`.

**Comment by andreasabel** [comment]
Why would type-checking-only suddenly work?  I thought our use of Template Haskell was a show stopper here, because then code is needed to type-check the rest? 
See discussion at #2038.

**Comment by Saizan** [comment]
@gallais I guess you could ```cabal configure``` without ```-fno-code``` first and then ```cabal build``` with ```-fno-code```.

However, even if Template Haskell is no longer a show stopper for ```-fno-code```[1] I still don't recommend it because the necessary object files are not cached properly (they get put in a temp directory and forgotten). Maybe we should open a ticket with ```ghc``` about that?

Personally I like to use ```ghcid``` for automatic reloading, and rely on ```cabal repl``` to run the "build":
```sh
ghcid -c "cabal repl --builddir=./dist-ghcid --ghc-options=-fobject-code --ghc-options=-O0 Agda"
```

[1] https://gitlab.haskell.org/ghc/ghc/issues/8025

**Comment by andreasabel** [comment]
AD: Soon coming to the computer in your office:
```
make type-check
```
Type-check the Agda source in a few seconds!

**Comment by gallais** [comment]
Beautiful!

**Comment by asr** [comment]
I'm getting an error on GHC 8.10.1. Before reporting the error I went to Travis to see the behaviour of `make type-check` on 8.10.1, but I couldn't find it. Please added `make type-check` to Travis.

**Comment by andreasabel** [comment]
`make type-check` is likely to _not_ work in a virginal clone of the repo.  I have also seen errors, but they went away after a `cabal configure` or `make quicker-install-bin` (I did not properly investigate this).

It would of course be nice if `make type-check` would work reliably _and_ not be slower.  If there is another 5 sec on some `cabal configure` each time I would be negative.

**Comment by UlfNorell** [comment]
The Travis job could do the proper setup before calling `make type-check`.

**Comment by nad** [comment]
I don't really see the point of having CI for `make type-check`.

I [fixed](https://github.com/agda/agda/commit/02cb18a317fe4ec94d2707b74676ad8bf6f4d3b9) one problem with `make type-check`: it used `cabal build` rather than `cabal v1-build`.

**Comment by asr** [comment]
@nad, please change `v1-build` by `$(CABAL_BUILD_CMD)`.

**Comment by nad** [comment]
OK.

**Comment by gallais** [comment]
~~Fixed by @andreasabel in 78cd3d81a691ad24c464e368ce9f603a0d15c127~~

Edit: ah, shoot. I hadn't seen that it had been closed & then re-opened.
Is it still broken on 8.10.1 after @nad's fix?

**Comment by andreasabel** [comment]
I am using it on a regular basis even with ghc 8.10 and it works for me.  There hasn't been a decision to test it on travis as well.
